src/main/java/org/apache/ibatis/migration/commands/BaseCommand.java

```diff
@@ -82,6 +83,15 @@
   protected BaseCommand(SelectedOptions selectedOptions) {
     this.options = selectedOptions;
     this.paths = selectedOptions.getPaths();
+    if (options.isQuiet()) {
+
+      this.printStream = new PrintStream(new OutputStream() {
+        @Override
+        public void write(int b) throws IOException {
+          // throw away output
+        }
```

Comment:
There's a setter for the printstream. This should use that but also it's also interesting that this shows a design flaw that we could set the printstream and override the quiet option without knowing it.

Reply:
Sure, there is a setter but setters are for outside classes. This is in the constructor - why would I call one of my own public functions?

Reply:
I now see what you're saying. `setPrintStream` and `quiet` are contradictory. I mean, technically you could have `quiet` doing something really shamless like putting if statements everywhere `if(!option.isQuiet()) printStream.println(...)` But that leave open a lot of places to make mistakes and just mis it. Plus, if it's quiet (even with these `if` statements, what's the point of setting the print stream?

The best thing to do is make them mutually exclusive and throw an error if both are set.

This could be done by declaring a specific instance of the quiet stream as:

```java
private static final PrintStream QUIET = new PrintStream(new OutputStream() {
        @Override
        public void write(int b) throws IOException {
          // throw away output
        }
      });
```

and then doing something like: 

```java
public void setPrintStream(PrintStream newStream) {
    if(this.printStream == QUIET) 
       throw new UnsupportedOpertionException("You can't have a quiet output and a set the output stream");

}
```